src/audio/copier/copier.c

```diff
@@ -740,6 +740,29 @@ static struct comp_dev *copier_new(const struct comp_driver *drv,
 	}
 
 	dev->pipeline = ipc_pipe->pipeline;
+	/* Calculation of the period in which the pipeline should be scheduled based on
+	 * the input buffer size.
+	 */
+	if (copier->gtw_cfg.dma_buffer_size > 0) {
+		/* Size of the single sample in bytes. */
+		uint32_t sample_size = copier->base.audio_fmt.channels_count *
+				       (copier->base.audio_fmt.depth / 8);
+		/* Size of the one millisecond of audio data in bytes. */
+		uint32_t one_ms = (sample_size * copier->base.audio_fmt.sampling_frequency) / 1000;
+		/* The number of milliseconds of data that will fit into the buffer. */
+		uint32_t ms_count = copier->gtw_cfg.dma_buffer_size / one_ms;
+
+
+		__ASSERT_NO_MSG(ms_count > 0);
+		if (ms_count == 0)
+			ms_count = 1;
```

Comment:
put 757-758 ahead of 756?

Reply:
since `ms_count` is unsigned, then if it's == 0, you need only one - either an assertion failure or a default value. Also note, that this will behave differently depending on whether asserts are enabled or not.

Reply:
I think this is a wrong use of an assertion then. I think asserts are used if ever during debugging to make sure that our assumptions are correct. That's why they're disabled at run-time. Because once we've checked that no cosmic rays develop an unhealthy affinity to our code, we can forget about them. OTOH, if that can indeed happen at run-time, then an assert is useless there and we need proper error checking.